File: app/src/main/java/com/example/mhmdreza_j/xproject/views/base_class/BaseFragment.java
package com.example.mhmdreza_j.xproject.views.base_class;

import android.app.job.JobScheduler;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import com.example.mhmdreza_j.xproject.views.main_page.MainActivity;

import org.greenrobot.eventbus.EventBus;

public class BaseFragment extends Fragment {
    public BaseFragment parentFragment = null;
    public BaseFragment nextFragment = null;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EventBus.getDefault().register(this);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        EventBus.getDefault().unregister(this);
    }

    public void onBackPressed() {
        if (getActivity() == null) return;
        getActivity().finish();
    }

    public void onNextPressed() {
        if (getActivity() == null || nextFragment == null) return;

        ((MainActivity) getActivity()).startFragment(nextFragment);
    }
}

File: app/src/main/java/com/example/mhmdreza_j/xproject/views/base_class/BaseFragment.kt
package com.example.mhmdreza_j.xproject.views.base_class

import androidx.fragment.app.Fragment
import android.widget.Toast
import com.example.mhmdreza_j.xproject.views.main_page.MainActivity
import com.example.mhmdreza_j.xproject.views.main_page.MainFragment

open class BaseFragment : Fragment() {
    var parentFragment: MainFragment? = null
    var nextFragment: BaseFragment? = null
    val mainActivity: MainActivity?
        get() {
            if (activity == null || activity !is MainActivity) {
                return null
            }
            return activity as MainActivity
        }

    open fun onBackPressed() {
        if (activity == null) return
        activity!!.finish()
    }

    fun onNextPressed() {
        if (activity == null || nextFragment == null) return
        mainActivity!!.startFragment(nextFragment!!)
    }


    fun showLoading() {
        if (activity == null) return
        mainActivity!!.showLoading()
    }

    fun hideLoading() {
        if (activity == null) return
        mainActivity!!.hideLoading()
    }

    protected fun toastMessage(message: String) {
        activity?.runOnUiThread {
            Toast.makeText(activity, message, Toast.LENGTH_SHORT).show()
        }
    }
}
